### PR TITLE
[XCBuildSupport] Implement Decodable for PIF types

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
This adds support for Decoding PIF types. The encoder uses
a CodingUserInfoKey named `preservePIFModelStructure` to preseve the
structure that makes it easier to decode the model vs encoding it for
xcbuild consumption.

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test self hosted

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test self hosted

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

File: Sources/XCBuildSupport/PIF.swift [hunk 1]
```diff
@@ -95,6 +107,12 @@ public enum PIF {
                 try container.encode(signature, forKey: "signature")
             }
         }
+
+        public required init(from decoder: Decoder) throws {
+            try super.init(from: decoder)
+            let container = try decoder.container(keyedBy: StringKey.self)
+            signature = try container.decode(String.self, forKey: "signature")
```
Comment (hartbit) [inline Sources/XCBuildSupport/PIF.swift:114]: Now that we use CodingKeys everywhere, perhaps we should also do so here?

**Pull request merged**